Decred Wallet/Features/Wallet Utils/Syncer.swift

```diff
@@ -208,6 +229,29 @@ class Syncer: NSObject, AppLifeCycleDelegate {
             self.networkLastActive = nil // Network is active at this point.
         }
     }
+
+    @objc private func reinstateBackgroundTask() {
+        if backgroundTask ==  .invalid {
+            registerBackgroundTask()
+        }
+    }
+
+    private func registerBackgroundTask() {
+        backgroundTask = UIApplication.shared.beginBackgroundTask { [weak self] in
+            self?.endBackgroundTask()
+        }
+        assert(backgroundTask != .invalid)
+    }
+
+    private func endBackgroundTask() {
+        print("Background task ended at: ", Date())
```

Comment:
This seems to be called multiple times, not really sure why. May be a good idea to check that `backgroundTask != .invalid` before proceeding here.

Reply:
> Just noticed that sync continues even after the background task is ended. See the log output here https://pastebin.com/Aqkkxp9g.
> 
> Also makes me wonder, what's the behaviour when the background task expires and is ended by the OS?

Should be fine now. Also, notice for every time the app goes into background we deregister any previous background task and register a new one.